src/main/java/uk/gov/hmcts/cp/cdk/batch/tasklet/GenerateAnswersTasklet.java

```diff
@@ -46,234 +46,268 @@ public class GenerateAnswersTasklet implements Tasklet {
     private final ObjectMapper objectMapper;
     private final DocumentInformationSummarisedApi documentInformationSummarisedApi;
     private final RetryTemplate retryTemplate;
+    private final DocumentIdResolver documentIdResolver;
 
     private static final String SQL_FIND_VERSION =
-            "SELECT version FROM answer_reservations " +
-                    "WHERE case_id=:case_id AND query_id=:query_id AND doc_id=:doc_id";
+        "SELECT version FROM answer_reservations " +
+        "WHERE case_id=:case_id AND query_id=:query_id AND doc_id=:doc_id";
 
     private static final String SQL_CREATE_OR_GET_VERSION =
-            "SELECT get_or_create_answer_version(:case_id,:query_id,:doc_id)";
+        "SELECT get_or_create_answer_version(:case_id,:query_id,:doc_id)";
 
     private static final String SQL_MARK_IN_PROGRESS =
-            "UPDATE answer_reservations " +
-                    "SET status='IN_PROGRESS', updated_at=NOW() " +
-                    "WHERE case_id=:case_id AND query_id=:query_id AND doc_id=:doc_id " +
-                    "AND status IN ('NEW','FAILED')";
+        "UPDATE answer_reservations " +
+        "SET status='IN_PROGRESS', updated_at=NOW() " +
+        "WHERE case_id=:case_id AND query_id=:query_id AND doc_id=:doc_id " +
+        "AND status IN ('NEW','FAILED')";
```

Comment:
do we need this additional check when we are updating the status? also, we can have. single parameterised SQL query for updating status and we can do away with the other two SQL_MARK_FAILED, SQL_MARK_DONE